### DISCUSSION TITLE
Question about needsOffscreenAlphaCompositing with Animated.View

### DISCUSSION THREAD

**Discussion by susonthapa** [software-mansion/react-native-reanimated#4887] [Q&A]
There is an open [issue](https://github.com/facebook/react-native/issues/23090#issuecomment-669157170) on React Native GitHub repository about Shadow and opacity on Android. The solution for the time being is to use `needsOffscreenAlphaCompositing`. Since this prop is quite expensive it is recommended to only use it during the animation. When I tried to do that with `Animated.View` it didn't work. If I set the prop to true it works but setting it to false and then enabling it later doesn't seem to work.

Here is a [repository](https://github.com/susonthapa/react-native-global-modal/tree/alpha-compositing-not-switching) that I was working on that has this issue. 

Video of the issue.

https://github.com/software-mansion/react-native-reanimated/assets/33973551/31cfd9e5-386f-4e99-b084-8da512f9ef57

As you can see the fading animation causes weird shadow issues on Android.

**Comment by Gaohaoyang** [comment]
same here

**Comment by derekvawdrey** [comment]
Same here, anyone have any solutions?

**Reply by MatiPl01** [reply]
Have you tried using `boxShadow` property? It is quite new and was added in RN 0.76. See [this](https://reactnative.dev/blog/2024/10/23/release-0.76-new-architecture#boxshadow) page for more details.
